src/live_view/live_view_server.js

```diff
@@ -0,0 +1,225 @@
+     *     "htmlContent": "<html><body> ....",
+     *     "screenshotIndex": 3
+     * }
+     * ```
+     *
+     * Screenshots are not served directly, only their index number
+     * which is used by client to retrieve the screenshot.
+     *
+     * @param {Page} page
+     * @return {Promise}
+     */
+    async serve(page) {
+        if (!this.hasClients()) return;
+        const snapshot = await this._makeSnapshot(page);
+        await this._pushSnapshot(snapshot);
+    }
+
+    /**
+     * @return {boolean}
+     */
+    get isRunning() {
+        return this._isRunning;
+    }
+
+    /**
+     * @return {boolean}
+     */
+    hasClients() {
+        return this.clientCount > 0;
+    }
+
+    /**
+     * Returns an absolute path to the screenshot with the given index.
+     * @param {number} screenshotIndex
+     * @return {string}
+     */
+    getScreenshotPath(screenshotIndex) {
+        return path.join(this.screenshotDirectoryPath, `${screenshotIndex}`);
+    }
+
+    async _makeSnapshot(page) {
+        const pageUrl = page.url();
+        log.info('Making live view snapshot.', { pageUrl });
+        const [htmlContent, screenshot] = await Promise.all([
+            page.content(),
+            page.screenshot(),
+        ]);
+
+        const screenshotIndex = this.lastScreenshotIndex++;
+
+        await writeFile(this.getScreenshotPath(screenshotIndex), screenshot);
+        if (screenshotIndex > this.maxScreenshotFiles - 1) {
+            this._deleteScreenshot(screenshotIndex - this.maxScreenshotFiles);
+        }
+
+        const snapshot = { pageUrl, htmlContent, screenshotIndex };
+        this.lastSnapshot = snapshot;
+        return snapshot;
+    }
+
+    async _pushSnapshot(snapshot) {
+        // Send new snapshot to clients
+        log.debug('Sending live view snapshot', { snapshot });
+        this.socketio.emit('snapshot', snapshot);
+    }
+
+    _deleteScreenshot(screenshotIndex) { // eslint-disable-line class-methods-use-this
+        unlink(this.getScreenshotPath(screenshotIndex))
+            .catch(err => log.exception(err, 'Cannot delete live view screenshot.'));
+    }
+
+    _setupHttpServer() {
+        const containerPort = process.env[ENV_VARS.CONTAINER_PORT] || LOCAL_ENV_VARS[ENV_VARS.CONTAINER_PORT];
+
+        this.port = parseInt(containerPort, 10);
+        if (!(this.port >= 0 && this.port <= 65535)) {
+            throw new Error('Cannot start LiveViewServer - invalid port specified by the '
+                + `${ENV_VARS.CONTAINER_PORT} environment variable (was "${containerPort}").`);
+        }
+        this.liveViewUrl = process.env[ENV_VARS.CONTAINER_URL] || LOCAL_ENV_VARS[ENV_VARS.CONTAINER_URL];
+
+        this.httpServer = http.createServer();
+        const app = express();
+
+        app.use('/', express.static(__dirname));
+
+        // Serves a JS file with the last snapshot, so the the client can immediately show something
+        app.get('/init-last-snapshot.js', (req, res) => {
```

Comment:
With `.js` in path I would expected this endpoint to return me a JS code not JSON API response.

Reply:
I would call it just `/last-snapshot`

Reply:
But it returns `text/javascript`.

Reply:
Ah sure, I missed that variable declaration inside.